### PR TITLE
feat: Added throughput support for root and EBS disks

### LINKED ISSUE #1444: Throughput support for gp3 disk type in launch configuration
## Description

Throughput support is added to aws_launch_configuration resource in provider [3.44.0](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#3440-june-03-2021)

## ⚠️ Note

> Before you submit an issue, please perform the following first:
> 
> 1. Remove the local `.terraform` directory (! ONLY if state is stored remotely, which hopefully you are following that best practice!): `rm -rf .terraform/`
> 2. Re-initialize the project root to pull down modules: `terraform init`
> 3. Re-attempt your terraform plan or apply and check if the issue still persists

When we provide `gp3` as `root_volume_type` in `workers_groups`, and also update the throughput for the gp3 disk, the changes doesn't get reflected. A fix/improvement would be to add `root_volume_throughput` config [here](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/workers.tf#L313) similar to the other configs

## Versions

 - Terraform: `0.14.11`

 - Provider(s): `3.44.0`

 - Module: `latest`

### Code Snippet to Reproduce

```
module "eks" {
  source          = "terraform-aws-modules/eks/aws"
  cluster_name    = local.cluster_name
  cluster_version = "1.18"
  subnets         = [<subnets>]

  tags = local.common_tags

  vpc_id = <vpc-id>

  worker_groups = [
    {
      name                          = "worker-group-1"
      instance_type                 = "t3.medium"
      additional_userdata           = local.additional_userdata
      additional_security_group_ids = [<sec groups>]
      root_volume_size              = 120
      root_volume_type              = "gp2"
      root_volume_throughput        = 350
      asg_min_size                  = 2
      asg_max_size                  = 5
      tags                          = flatten(local.worker_tags)
    },
  ]
}
```

## Expected behavior

Throughput configuration of `350 MiB/s` should have reflected in the provisioned volume

## Actual behavior

Throughput value is the default `125 MiB/s`

### PR DESCRIPTION
**Pull request by junaid-ali**
# PR o'clock

## Description

Fixes #1444
When `gp3` disk type is used, throughput should be configurable for an ASG launch configuration. This PR adds that support.

Example usage:
```terraform
module "eks" {
  ...

  worker_groups_launch_template = [
    {
      name                   = "worker-group-gp3"
      instance_type          = "t3.small"
      asg_desired_capacity   = 1
      public_ip              = true
      root_volume_size       = 150
      root_volume_type       = "gp3"
      root_volume_throughput = 300 # setting throughput to 300 MiB/s
    },
  ]
}
```

### Checklist

~- [ ] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation~

### CODE REVIEW

**Comment by junaid-ali** [comment]
@daroga0002 any chance if you can have a look at this?

**Comment by junaid-ali** [comment]
@barryib @max-rocket-internet @daroga0002, could you please have a look at this XS PR?

**Comment by antonbabenko** [comment]
@junaid-ali Please update the PR and add `throughput` for EBS volumes and update code in `examples/launch_templates` to show this feature in action, too.

**Comment by junaid-ali** [comment]
thanks @antonbabenko , @daroga0002 for the review, I'll add the example

**Comment by junaid-ali** [comment]
@antonbabenko added the example usage to `examples/launch_templates`, and PR description as well. Attached is the plan output:
Screenshot 2021-09-01 at 12 10 26

Could you please have a look at it again?

**Comment by antonbabenko** [comment]
I don't see changes for `throughput` inside of `ebs_block_device` block [here](https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1445/files#diff-abc45b07ed4bab6d21313623302e9685703e3b4ad4a981fd57fe91c4c70095edR315) and not in examples. Have you pushed all the changes?

File: examples/launch_templates/main.tf [hunk 1]
```diff
@@ -73,5 +73,14 @@ module "eks" {
...
     },
+    {
+      name                   = "worker-group-4"
+      instance_type          = "t3.small"
+      asg_desired_capacity   = 1
+      public_ip              = true
+      root_volume_size       = 150
+      root_volume_type       = "gp3"
+      root_volume_throughput = 300
+    },
```
Comment (junaid-ali) [inline examples/launch_templates/main.tf:84]: @antonbabenko From what I understood with your first comment, you are expecting an example like this at `examples/launch_templates/main.tf`, right? Please correct me if I misunderstood.

> I don't see changes for throughput inside of ebs_block_device block [here](https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1445/files#diff-abc45b07ed4bab6d21313623302e9685703e3b4ad4a981fd57fe91c4c70095edR315)

The 'here' link somehow only show me the PR diff page, and isn't highlighting anything.

Comment (antonbabenko) [reply examples/launch_templates/main.tf:84]: The link works only if you expand the PR a bit.

I meant that `dynamic "ebs_block_device"` block on line 315 has to have a `throughput` in content, too.

Once there is a `throughput` in `ebs_block_device`, update file `examples/launch_templates/main.tf` and add `additional_ebs_volumes` with some data including the new argument you add.

Comment (junaid-ali) [reply examples/launch_templates/main.tf]: Thanks @antonbabenko, fixed now.

For launch templates, the config is already there: https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/workers_launch_template.tf#L505

**Comment by junaid-ali** [comment]
Version updated in the README

**Comment by antonbabenko** [comment]
The code looks good, but I want to verify it a bit more during this week.

**Comment by antonbabenko** [comment]
Thanks @junaid-ali !

[v17.7.0](https://github.com/terraform-aws-modules/terraform-aws-eks/releases/v17.7.0) has been just released.

PS: I am unsure whether we need to implement the same changes in `modules/node_groups`. WDYT @daroga0002 ?

**Comment by daroga0002** [comment]
seems issue for this enhancement was created:
https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1567

**Pull request merged**